cli/src/dev/hot_reload.rs

```diff
@@ -107,7 +107,10 @@ pub(crate) async fn hot_reload(
         subgraph_watcher.stop();
 
         let config = match Config::load(&config_path) {
-            Ok(config) => config,
+            Ok(mut config) => {
+                config.graph.introspection = Some(true);
```

Comment:
This was a bug. The first gateway config would always get enabled introspection, but not anymore as the config gets reloaded.

Reply:
Will add that to the changelog